### ISSUE TITLE
Train yolov5 with rectangle image

### ISSUE THREAD

**Issue by minhdov** [ultralytics/yolov5#8195]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

When training yolov5 with rectangle image with --imgsize is 640. I see that yolov5 always upscales the small image (i.e 320x254) to the 640 range that I don't want to happen. Could we disable image upscale in yolov5 training? So that, the images have a size larger than 640, we resize to the 640 range, images have a size smaller than 640, we keep the original image size

**Comment by glenn-jocher** [comment]
@minhdousd if you want to train at 320 then set --img 320

**Comment by minhdov** [comment]
Thanks for your prompt response. My test set has different image sizes (120x240, 155x160, 320x450, 640x480). I want to do inference with the original input image size to accelerate the running time. Can I set --img 640 for training and do not scale up (i.e 120x240, 155x160) to the 640 range?

**Comment by glenn-jocher** [comment]
@minhdousd we don't have a way to run inference at native image sizes other than specifying image size directly on pytorch hub models, i.e. `results = model(im, size=320)`.

YOLOv5 🚀 [PyTorch Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading) models allow for simple model loading and inference in a pure python environment without using `detect.py`. 

### Simple Inference Example

This example loads a pretrained YOLOv5s model from PyTorch Hub as `model` and passes an image for inference. `'yolov5s'` is the YOLOv5 'small' model. For details on all available models please see the [README](https://github.com/ultralytics/yolov5#pretrained-checkpoints). Custom models can also be loaded, including custom trained PyTorch models and their [exported](https://docs.ultralytics.com/yolov5/tutorials/model_export) variants, i.e. ONNX, TensorRT, TensorFlow, OpenVINO YOLOv5 models.

```python
import torch

# Model
model = torch.hub.load('ultralytics/yolov5', 'yolov5s')  # yolov5n - yolov5x6 official model
#                                            'custom', 'path/to/best.pt')  # custom model

# Images
im = 'https://ultralytics.com/images/zidane.jpg'  # or file, Path, URL, PIL, OpenCV, numpy, list

# Inference
results = model(im)

# Results
results.print()  # or .show(), .save(), .crop(), .pandas(), etc.

results.xyxy[0]  # im predictions (tensor)
results.pandas().xyxy[0]  # im predictions (pandas)
#      xmin    ymin    xmax   ymax  confidence  class    name
# 0  749.50   43.50  1148.0  704.5    0.874023      0  person
# 2  114.75  195.75  1095.0  708.0    0.624512      0  person
# 3  986.00  304.00  1028.0  420.0    0.286865     27     tie
```
<img width="550" src="https://user-images.githubusercontent.com/26833433/149679662-b2021c78-59b2-4f52-b768-1e5161537c85.png">

See [YOLOv5 PyTorch Hub Tutorial](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading) for details.

Good luck 🍀 and let us know if you have any other questions!

**Comment by minhdov** [comment]
Does the yolov5 support training the model with native image size (padded to be multiple of 32) without upscale to a certain range like (--640)? When I disable scaleup in letterbox function for training, the evaluation code happens problem with tensor size

![image](https://user-images.githubusercontent.com/39869815/173513482-ce436616-a684-4301-95c0-390ce78306ef.png)

**Comment by glenn-jocher** [comment]
@minhdousd no, there is no official native size support. I would simply find the closest image size that corresponds to average image size pass that.
